Mainline MIPS R6 64-bit (Little Endian) Port

MingcongBai opened the pull request:
Topic Description
-----------------

This topic merges the CIP-maintained MIPS R6 64-bit (Little Endian) port (or `mips64r6el`) as a secondary AOSC OS mainline port.

Apart from introducing a new port, this pull request also contains full stage2 recipes, benefiting future ports. Please note that this port does not contain or enable MSA (MIPS SIMD Architecture), which could benefit actual hardware. Should CIP decide that all MIPS R6 devices be MSA-enabled, we will open another pull request to merge those changes.

This pull request also introduces "MIPS R6 64-bit (Little Endian)" as a secondary architecture. Future update builds for this port will be handled by CIP employees and interns.

Maintainers: @Cyanoxygen @MingcongBai

Package(s) Affected
-------------------

For other ports, with a few exceptions, this pull request does not affect package version numbers currently in the mainline tree. The following were unfortunately affected and will therefore need to be built and tested on other mainline architectures.


```
cgdb/now 0.8.0 mips64r6el [installed,local]
exfat-utils/now 1.3.0-1 mips64r6el [installed,local]
expat/now 2.4.9 mips64r6el [installed,local]
libpng/now 1.6.37-2 mips64r6el [installed,local]
mdadm/now 4.1-5 mips64r6el [installed,local]
openssh/now 8.8p1 mips64r6el [installed,local]
polkit/now 1:121 mips64r6el [installed,local]
python-3/now 3.10.8-1 mips64r6el [installed,local]
xfsprogs/now 5.14.2-1 mips64r6el [installed,local]
```

Security Update?
----------------

No

Build Order
-----------

```
texinfo ocaml findlib bzip2 python-3 llvm libidn ca-certs gcc util-linux cryptsetup systemd polkit accountsservice apt autofs mdadm mtd-utils osinfo-db-tools quota-tools thin-provisioning-tools xfsprogs desktop-base bird server-base talloc tevent tdb ldb libarchive acbs cgdb ftjam ladspa-sdk unzip docbook-dtd libotf m17n-db m17n-lib emacs gcr avahi jack libvdpau aom samba sdl2 pulseaudio ffmpeg libpng gdk-pixbuf gtk-4 nodejs plymouth clucene dbus-glib glib-networking hidapi libclc libcork libgxps libiptcdata libmcrypt libnice libsoup libtasn1 libwmf mbedtls minizip nss qt-4 qt-6 yajl raptor2 serf spandsp zita-resampler aosc-artwork unicode-ucd libcaca gstreamer cifs-utils postfix ppp uw-imap dbus-deviation numpy py3c pyqt5-sip sip wheel openblas arch-chroot genfstab pacstrap hdparm iotop kmscon neofetch p7zip rhash subversion squid
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Cyanoxygen committed:
glib: add mips64r6el specific recipe and quirks

- Disable tests due to failure on producing test resources, see
  https://gitlab.gnome.org/GNOME/glib/-/issues/2768.

- Disable gtk-doc generation due to missing references.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libidn: add gtk-doc as build dependency

- Also lint code

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
python-3: mips64r6el: fix multiarch detection

By hard coding host triple in configure script.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
bzip2: create target directory before installing pkg-config file

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cryptsetup: update to 2.5.0

This update is required in order to build systemd 251 due to header
file incompatiblity.

Cyanoxygen committed:
brltty: export LDEMULATION="elf64ltsmip" to fix target emulation errors during link time

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
apt: update to 2.5.4

- Drop patch to hack po4a detection.
- (mips64r6el) drop apt-gen-list dep, which depends on unavailable Rust.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cryptsetup: fix build for 2.5.0

- Drop invalid option: --enable-cryptsetup-reencrypt.
- Add asciidoctor build dep.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gamin: (no build) fix source link

No change in checksum, so no need to rebuild this package or bump REL.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
quota-tools: make rpcsvc-proto file removal conditional

These files did not end up in PKGDIR during our test build for mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
systemd: fix build

- Do not install tmpfiles.d/legacy.conf as it no longer exists upstream.
- Add jinja2 build dep.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
thin-provisioning-tools: (mips64r6el) disable Rust tools

- Disable Rust tools for missing toolchain support on mips64r6el.
- Add a note to re-enable Rust tools on riscv64.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
admin-base: (mips64r6el) drop atm dep

ATM requires Rust to build. Rust is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
boot-base: (mips64r6el) drop efibootmgr dep

efivars is not yet supported on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
desktop-base: add missing build deps

- Add imagemagick, lxml, and pyyaml build deps.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
network-base: (mips64r6el) drop clash, v2ray deps

Go is not yet supported on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
server-base: (mips64r6el) drop v2ray dep

Go is not yet supported on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
util-base: (mips64r6el) drop bat, fd, ripgrep deps

These tools require Rust to build, which is not yet available for mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ldb: update to 2.6.1

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
tdb: update to 1.4.7

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
acbs: update to 20221105

This update introduces .stage2 support.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ftjam: fix versiont to 2.5.3~rc2

- Bump EPOCH for version calibration.
- Fix source link as upstream source disappeared.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ladspa-sdk: update to 1.17

- This update fixes build on mips64r6el.
- Add libsndfile dep.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ghostscript: (mips64r6el) disable parallelism

Parallel build fails on mips64r6el (and seemingly only on mips64r6el, when
using Qemu user-mode emulation), as build system complains about missing
objects during build-time.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
emacs: update to 28.2

Original stable version (27.2) fails to build with current system dependencies.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gcr: add libsecret dep

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ibus: (mips64r6el) drop emojione dep

emojione requires Node.js to build, which is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
m17n-db: drop unneeded chmod a-s on SRCDIR

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
m17n-lib: drop unneeded chmod a-s on SRCDIR

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
nodejs: mark FAIL_ARCH

- Node.js is not yet available on mips64r6el due to a segfault in V8. A fix was
  attempted by Jiaxun Yang (FlyGoat), but it is now unusable again.
- Upstream bug is not to be opened yet, per FlyGoat's suggestion - wait until
  a downstream fix is available and tested.
- Lint scripts in accordance with the Styling Manual.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
kernel-tools: (mips64r6el) set default LD emulation to fix linkage errors

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
linux-kernel: (mips64r6el) add production-ready configuration

- Current kernel config will work with both Boston boards and Qemu's emulated
  "virt" machine.
- Install both .ltb and ELF kernel images for good measures (buggy Qemu may
  result in the ELF kernel image failing to boot on emulated Boston boards).

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
openfwwf: (no build) fix source link

No change in source checksum, no rebuild necessary.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
plymouth: (upstream patch) fix build against Glibc 2.36

Extraneous <linux/fs.h> would result in build failure in libply/ply-utils.c
against Glibc 2.36, backport a patch from the upstream to fix build.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
clucene: update to 2.3.3.4+git20130812

- Following snapshot used by Fedora and sync their patches.
- Use cmakeninja ABTYPE.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
dbus-glib: add gtk-doc build dep

Add gtk-doc build dep to fix reconfiguration.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
hidapi: update to 0.12.0

- Upstream moved under libusb.
- Original version in stable (0.8.0rc1) fails to build against current system
  dependencies.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libavif: (mips64r6el) drop rav1e dep, disable rav1e codec

rav1e requires Rust to build, which is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libclc: update to 14.0.6

- Version in stable (20181128-2) fails to build against current system
  dependencies.
- Add spirv-llvm-translator, spirv-tools build deps.
- Bump EPOCH for version format change.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libcork: (Debian patch) update to 1.0.0~rc3

- Switch to cmakeninja build type.
- Introduce a Debian patch to fix documentation build with Sphinx >= 3.1.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libheif: (mips64r6el) drop rav1e dep

rav1e requires Rust to build, which is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libmcrypt: drop extraneous autoreconf

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libsodium: (mips64r6el) reference AUTOTOOLS_AFTER__NONX86

Declare that mips64r6el is not an x86 architecture, duh.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libsoup: add samba build dep

This satifies the dependency requirement for NTLM support.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libtasn1: add gtk-doc build dep

This fixes a build-time failure when Autobuild3 attempts to regenerate
autotools scripts.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libvdpau: add doxygen, graphviz build deps for documentation generation

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libwmf: fix build with newer FreeType

- Add freetype build dep.
- Append /usr/include/freetype2 to header search directories.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
minizip: fix a typo, PKGEC => PKGSEC

Also fix source link as upstream source disappeared.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
opencv: (mips64r6el) drop OpenJDK/Java dependencies

OpenJDK/Java is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
qt-4: (Fedora patches) various adaptations and fixes

- Sync patches from Fedora.
- Re-introduce MariaDB/MySQL dependency and feature.
- Use bundled Clucene to prevent build failures, as system dependency no longer
  maintains API compatibility with Qt 4.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
qt-5: declare mips64r6el as no-WebEngine architecture

Also drop nodejs build dep for no-WebEngine architectures. Node.js is not yet
available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
qt-6: various fixes and adaptations

- Add libxkbcommon, vulkan deps.
- Declare mips64r6el as no-WebEngine architecture.
- Drop nodejs build dep for no-WebEngine architectures. Node.js is not yet
  available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
raptor2: disable gtk-doc

As of gtk-doc 1.26, gtkdoc-mktmpl has been removed. This tool is required by
Raptor2 to build gtk-doc documentations.

Upstream has yet to come up with a workable fix. Disable this feature for now.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
sdl2: update to 2.24.2

This update fixes build against newer Wayland.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
serf: drop old Python 3 adaptation patch

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
spandsp: add docbook-xsl, libxslt build deps

These build dependencies are required for generating documentations
with Doxygen.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
talloc: update to 2.3.4

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
tevent: update to 0.13.0

Add cmocka build dep.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
wayland: (mips64r6el) disable documentation

FIXME: dot SIGILLs at the following command ...

/usr/bin/dot -Tpng -odoc/doxygen/xml/wayland-architecture.png ../doc/doxygen/dot/wayland-architecture.gv

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
wpebackend-fdo: (mips64r6el) drop hotdoc build dep

hotdoc will eventually depend on Rust, which is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
yajl: use Git source

GitHub-generated tarballs are not reliable as they could be re-rolled over
time, resulting in checksum changes.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
zita-resampler: update to 1.8.0

- (mips64r6el) Introduce a patch to prevent SSE2 macros from being linked.
- Hack LDFLAGS to prevent bootstrapping.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
aosc-artwork: add lxml, pyyaml build deps

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
unicode-ucd: create a compatibility symlink for emoji-test.txt

/usr/share/unicode/emoji/emoji-test.txt => emoji-tests.txt, which is needed to
build ibus. This could be a change introduced with Unicode UCD version 14.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ffmpeg: various fixes and adaptations

- Introduce mips64r6el build spec (mips64r6 CPU, no SIMD enabled).
- Use ab_match_arch macros over old architecture matching expressions,
  where applicable.
- Add a patch to enable mips64r6 generic options.
- Drop deprecated qt-faststart installation routine.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gstreamer: various fixes and adaptations

- Add openjpeg dep.
- Add graphviz build dep.
- Drop additional potentially bundled files.
- (mips64r6el) Drop zbar dep, which requires not-yet-available Rust support.
- (mips64r6el) Drop hotdoc build dep, which requires not-yet-available Rust
  support.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
jack: add zita-alsa-pcmi, zita-resampler deps

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libbluray: (mips64r6el) drop openjdk/apache-ant build deps

OpenJDK/Java is not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libcaca: add doxygen dep, disable shadow build

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
pulseaudio: various fixes and adaptations

- Drop unneeded alternatives.
- Add pulseaudio to own build dep in full recipe.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
x264: (mips64r6el) append -mmsa to fix build

FIXME: Although MSA should be supported by all I6400/6600 cores, the P6600
core, which is also potentially going to be used in CIP products, does not
necessarily come with its modular MSA pipelines.

FIXME: CIP may in the end enforce MSA for all distribution releases, awaiting
final decision.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
avahi: various fixes and adaptations

- (mips64r6el) Drop twisted dep, which will depend on not-yet-available Rust
  support.
- (mips64r6el) De-duplicate AUTOTOOLS_AFTER__MIPS64R6EL by referencing the
  generic AUTOTOOLS_AFTER variable.
- Add doxygen build dep.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
bird: update to 2.0.10

This update fixes build against GCC >= 10.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cifs-utils: move /sbin => /usr/bin to comply with Autobuild3 QA requirements

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
networkmanager: (mips64r6el) disable LTO

FIXME: Somehow, LTO depletes system RAM when building with Qemu user-mode
emulation, risking a full sytem lock-up (including the host).

Investigation pending.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
postfix: update to 3.7.3

This update fixes build against Glibc >= 2.36, Linux Kernel >= 6.x (host).

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
samba: update to 4.17.2

- This update fixes build against current system dependencies.
- Add perl-json dep.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
uw-imap: drop unneeded chmod a-s on SRCDIR

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
findlib: fix build against OCaml 4.14.0

FIXME: Also disable LTO on mips64r6el, which results in build failure.

Investigation pending.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ocaml: (mips64r6el) disable LTO

Investigation pending.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
dbus-deviation: add pip build dep

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
numpy: specify python ABTYPE

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
py3c: use self-defined build script

py3c requires building from outside of the source tree, where no additional
files and directories are found in its source root.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
sip: add python-build, python-installer, wheel build deps

python-build wheel python-installer

Cyanoxygen committed:
wheel: (mips64r6el) drop keyring dep

- keyring => cryptography => setuptools-rust => rustc, but Rust support is not
yet available on mips64r6el. Replace with python-2, python-3 dependencies.
- Mark package as arch-dependent, add ABSPLITDBG=0.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
openblas: update to 0.3.21+git20221105

This update is necessary for mips64r6el support (MIPS64_GENERIC).

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
texlive: (mips64r6el) drop Java/LuaJIT deps

OpenJDK/Java and LuaJIT are not yet available on mips64r6el.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
arch-install-scripts: update mirrorlist.pacstrap checksum

FIXME: Track file changes in a Git repository or local mirror.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
kmscon: update to 8+git20221110

This update fixes build on mips64r6el, as it prevents linkage by directly invoking
ld, producing incompatible linker objects.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
p7zip: drop File Manager

Strip out wxWidgets dependencies and re-unify stage2 recipes with full recipes.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
rhash: drop unneeded beyond script

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
util-linux: disable raw character device support

FIXME: raw(8) is no longer available as of Linux API headers 5.14.

Ref: https://src.fedoraproject.org/rpms/util-linux/blob/rawhide/f/util-linux.spec#_965

Ref: Commit in torvalds/linux.git ...
     https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=603e4922f1c81fc2ed3a87b4f91a8d3aafc7e093

Disable raw for all architectures, because linux/raw.h was deprecated since commit:
603e4922f1c81fc2ed3a87b4f91a8d3aafc7e093.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
subversion: various fixes and adaptations

- Disable parallelism, which can break installation.
- (Fedora patch) Fix installation-time linkage.

Cyanoxygen committed:
accountsservice: (upstream patch, adapted) fix build against Meson >= 0.60

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
autofs: (Fedora, upstream patches) fix build with Glibc >=2.36

- Sync patches from Fedora.
- Devise a patch to fix build against Glibc >= 2.36 by dropping extraneous
  <linux/fs.h> includes.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
accountsservice: bump REL for build fix

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gtk-4: various fixes and adaptations

- (mips64r6el) Specify ld emulation target to avoid linkage failure.
- Use compiler to generate resource binaries.

Ref: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2923/diffs

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
glib-networking: skip broken TLS tests

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libotf: (Arch Linux patch) use pkg-config data to determine FreeType configurations

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
openh264: (mips64r6el) disable MIPS assembly for r6

Currently, the MIPS assembly core included in openh264 targets r2 (and
Loongson 3, more specifically). This is incompatible with MIPS r6.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ca-carts: update make-ca.sh script to version 20170514

    This update makes the make-ca script produce CA certificate bundle.
    Also this update supports DESTDIR, adjust build script to reflect
    this update.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
python-3: (CIP calibration) update to 3.10.8-1

We messed up. As we attempted to calibrate versions in step 0, the following
packages' downgrade was overridden by pre-existing packages already uploaded
to the CIP repository server. These changes will be merged back to AOSC-Dev,
potentially causing merging problems with other topics.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mdadm: (CIP calibration) update to 4.1-5

We messed up. As we attempted to calibrate versions in step 0, the following
packages' downgrade was overridden by pre-existing packages already uploaded
to the CIP repository server. These changes will be merged back to AOSC-Dev,
potentially causing merging problems with other topics.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
polkit: (CIP calibration) update to 121

We messed up. As we attempted to calibrate versions in step 0, the following
packages' downgrade was overridden by pre-existing packages already uploaded
to the CIP repository server. These changes will be merged back to AOSC-Dev,
potentially causing merging problems with other topics.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
xfsprogs: (CIP calibration) update to 5.14.2-1

We messed up. As we attempted to calibrate versions in step 0, the following
packages' downgrade was overridden by pre-existing packages already uploaded
to the CIP repository server. These changes will be merged back to AOSC-Dev,
potentially causing merging problems with other topics.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cgdb: (CIP calibration) update to 0.8.0

We messed up. As we attempted to calibrate versions in step 0, the following
packages' downgrade was overridden by pre-existing packages already uploaded
to the CIP repository server. These changes will be merged back to AOSC-Dev,
potentially causing merging problems with other topics.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
linux-kernel: (mips64r6el) improve Boston config, backport patches

- 1001-HACK-cevt-r4k-Don-t-call-event_handler-if-it-s-not-r.patch.mips64r6el

  HACK: cevt-r4k: Don't call event_handler if it's not registered

  After SMP secondary bringup, there will be a very short peroid that IP7 is
  enabled but event_handler is not registered.

- 1002-MIPS-jump_label-Fix-compat-branch-range-check.patch.mips64r6el

  MIPS: jump_label: Fix compat branch range check

  Cast upper bound of branch range to long to do signed compare,
  avoid negative offset trigger this warning.

  Fixes: 9b6584e35f40 ("MIPS: jump_label: Use compact branches for >= r6")

- (mips64r6el) Improve Boston configuration to better conform to its
  integrated peripherals.

Reference: https://github.com/torvalds/linux/blob/6e62a888029b4adbbc55894eaa8b929221113948/arch/mips/configs/generic/board-boston.config

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
openssl: (mips64r6el) set target to linux64-generic64

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
hfsprogs: fix source link

Apple Open Source no longer ships source tarballs, archive it at our
aosc-repack directory.

No rebuild necessary due to lack of checksum changes.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
osinfo-db-tools: add libsoup dep

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
libgxps: add gtk-doc build dep

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
libiptcdata: add gtk-doc build dep

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
pyqt5-sip: set ABTYPE to python

Otherwise, Autobuild3 defaults to pep517, which fails to build this package.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
r: drop openjdk dep for loongson3, mips64r6el, riscv64

FIXME: OpenJDK is available for loongson3, consider re-enabling.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Commit:
systemtap: fix source link

Upstream source location changed. No rebuild necessary due to lack of checksum
changes.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

MingcongBai committed:
meta: add "MIPS R6 64-bit (Little Endian)" (mips64r6el) to pull request template

Icenowy commented:
Maybe the list of packages need to be updated on other architectures needs to be longer, e.g. https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/commits/c90403a23ed877e3950b43330726b7825f12fcb2 for ppp, https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/commits/122857fd338d5f32bf43a8cc80d9fbcddf1427bd for postfix, etc.

(My criteria is that all changes that are not marked architecture-specific need to be propagated to all architectures.)

MingcongBai commented:
@Cyanoxygen Please help address this.

Cyanoxygen commented:
List of packages that requires rebuild
--------

Here's the list of packages which requires rebuild and a successful test:

- llvm [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-c27b8a2bb39e9f06668a91847b8af1b475f63b30279459c4c4499abb4320b610)
- libidn [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-6b80014cdcd13d2c3e1d3831f67c43b73d69c37ef2aa8d13f3b6b4d59814b6ce)
- ca-certs [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-9e746715de0bcbee62f2fcd477fb7006a51612441d3f35096d439e9663497b49)
- python-3 (REL bump)
- bzip2 [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-0cb1028caa46ab9e1bfe1689442d3f5711b725a4cb84d038f7c336eaaff81039)
- texinfo (new patch, REL bump)
- gcc (removed dependency zstd)
- accountsservice (new patch, REL bump)
- apt (1 removed patch, version update)
- autofs (Bunch of patches intoduced)
- cryptsetup [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-7c974d85bd9e65da8b01eea0522e916f5d8e182d130f13e7bca997707ce0252f)
- mdadm (REL bump)
- mtd-utils (REL bump)
- osinfo-db-tools (added libsoup as dependency)
- polkit (updated to v121, removed two patches)
- quota-tools (added beyond script)
- systemd (updated to 251.6)
- thin-provisioning-tools (linted beyond script, NO REL BUMP)
- xfsprogs (REL bump)
- desktop-base (added pyyaml, lxml, imagemagick as build dependency)
- python-base (no longer noarch)
- server-base (no longer noarch)
- ldb (updated to 2.6.1)
- tdb (updated to 1.4.7)
- ~~acbs (in survey)~~
- cgdb (updated to 0.8.0)
- ftjam (added epoch)
- ladspa-sdk (updated and added libsndfile as depednency)
- docbook-dtd [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-129e8b2f1a8a262ee768787dff22352a7a509254ef67025555417547cf6b48b3)
- emacs (updated to 28.2)
- gcr (added gobject-introspectoin as dependency)
- gtk-4 (new patch introduced)
- m17n-db, m17n-lib
- nodejs
- plymouth (new patch introduced)
- aom (REL bump)
- clucene (bunch of new patches introduced)
- dbus-glib (added gtk-doc as build dependency)
- gdk-pixbuf (added shared-mime-info as dependency)
- glib-networking (REL bump, new patch introduced)
- hidapi (updated to 0.12.0)
- libarchive (new patch introduced)
- libclc (updated to 1:14.0.6)
- libcork (updated to 1.0.0~rc3)
- libgxps (REL bump, added gtk-doc as build dependency)
- libiptcdata (REL bump, added gtk-doc as build de[endency)
- libmcrypt (removed prepare script)
- libnice (REL bump)
- libotf (REL bump, new patch introduced)
- libpng (added libpng as build dependency, REL NOT BUMPED)
- libsoup (REL bump, added samba as build dependency)
- libtasn1
- libvdpau (REL bump)
- libwmf [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-8f7a04939c58faa1df0f552350220d8d5e38391b0ecd5f792b2a9547da52dc16) (added freepype as dependency)
- mbedtls
- minizip (REL bump, fixed typo in script)
- nss
- qt-4 (bunch of patched introduced, added mariadb as build dependency)
- qt-6 [diff](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4351/files?file-filters%5B%5D=.deferred&file-filters%5B%5D=.diff&file-filters%5B%5D=.mips64r6el&file-filters%5B%5D=.patch&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-45243cf992db4583fdd5352ecd79ccfabac31582d4f2c4c57755f29c5e7751ae)
- raptor2 (REL bump, some changes in defines)
- sdl2 (updaed to 2.24.2)
- serf (patch removed)
- spandsp (added docbook-xsl and libxslt as build dependency)
- talloc (updated to 2.3.4)
- tevent (added cmocka as dependency)
- yajl (switched to git tag, REL bump)
- zita-sampler
- aosc-artwork (added pyyaml, lxml as build dependency)
- unicode-ucd (added beyond script)
- ffmpeg (REL bump)
- gstreamer (modified beyond, defines, REL bump)
- jack (modified build, defined, REL bump)
- libcaca (added doxygen as build dependency)
- pulseaudio (added itself as build dependency)
- avahi (added doxygen as build dependency)
- bird (updaed to 2.0.10)
- cifs-utils (modified beyond script)
- postfix (patch related, REL bump)
- ppp (modified build script)
- samba (added perl-json as dependency)
- uw-imap (removed beyond (which was removing stickty bit from SRCDIR))
- findlib (modified build scrpit)
- ocaml (modified CFLAGS)
- dbus-deviation (added pip as dependency)
- numpy (added ABTYPE=python, REL bump)
- py3c (modified build scrpit, REL bump)
- pyqt5-sip (added ABTYPE=python, REL bump)
- sip (added BUILDDEP)
- wheel (no longer noarch)
- openblas (REL bump)
- arch-install-scripts (REL bump)
- hdparm (removed patches)
- iotop (removed beyond)
- kmscon (added prepare script, updated to 8+git20221110)
- neofetch
- p7zip (removed wxgtk3 dependency, modified build script)
- rhash (removed beyond, REL bump)
- unzip (bumped EPOCH)
- util-linux (changes in build options, REL bump)
- subversion
- squid: REL bump

Cyanoxygen committed:
texinfo: introduce patch to make gpm happy

- Also change the permission of postinst script

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

(Lost commit recovery)

Cyanoxygen committed:
libpng: add itself as dependency (lost commit recovery)

- Add libpng to build dep for main recipe.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
m17n-db: bump REL

- Finishes up commit 30eddce631f23a8d7ebf6bee236ac5458e2fc68c .

Cyanoxygen committed:
m17n-lib: bump REL

- Finishes up cc1549f342af8de801c64c323cb8d54db5c6d848 .

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
nodejs: bump REL

- Finishes up commit dc7971bf2486c520ae53a362855a8fb5615bdf4a.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
dbus-glib: bump REL

- Finishes up commit af1ba12d6dba38c900b6848e942d0019fe26b0da.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mbedtls: bump REL

- Finishes up commit 32dd629a1cddc6762f0e33163ee4541cf336041f.

Cyanoxygen committed:
nss: bump REL

- Finishes up commit fe1ea428639b217112c96900e3ddb1211587e284:
  Also removed XCFLAGS line.

Cyanoxygen committed:
qt-4: bump REL

- Finishes up commit 213371322c8f37d396f30e1421ad87b70b9f83a9.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ppp: bump REL

- Finishes up commit 0aaa48c2ebef5c87edc317c5ec34b120236c19aa.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
uw-imap: bump REL

- Finishes up commit 757a8f16870498871532de29e6a8f1308c8f1955.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
entrypoints: bump REL

- Finishes up commit 39039819312e84e2dc73d6584be28226db7f4b9c.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ptyprocess: bump REL

- Finishes up commit 6c03aa14162f389c59bf44cca70345fe24393b6f.

Cyanoxygen committed:
subversion: bump REL

- Finishes up commit ec24310d4d5388a0e8a7947aae9d734b73609823.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
python-3: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
accountsservice: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
bzip2: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gcc: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
llvm: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ca-certs: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
apt: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
texinfo: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
autofs: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libidn: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ldb: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
desktop-base: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cgdb: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ladspa-sdk: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
server-base: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ftjam: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
systemd: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
thin-provisioning-tools: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
polkit: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
xfsprogs: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
docbook-dtd: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mtd-utils: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
acbs: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
quota-tools: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
tdb: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cryptsetup: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mdadm: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
emacs: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
osinfo-db-tools: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libcork: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libgxps: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
m17n-db: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libiptcdata: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
glib-networking: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
hidapi: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libclc: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libmcrypt: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gcr: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gtk-4: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gdk-pixbuf: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libnice: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
aom: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
clucene: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
plymouth: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
dbus-glib: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
m17n-lib: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
nodejs: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libarchive: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libtasn1: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
qt-6: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
nss: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
sdl2: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libvdpau: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
yajl: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
tevent: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
talloc: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
serf: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
raptor2: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
spandsp: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
minizip: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
qt-4: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
zita-resampler: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libsoup: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libotf: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mbedtls: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libpng: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libwmf: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
avahi: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
findlib: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
cifs-utils: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
numpy: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
py3c: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ppp: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
aosc-artwork: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ocaml: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
jack: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ffmpeg: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
pulseaudio: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
pyqt5-sip: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
libcaca: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gstreamer: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
bird: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
postfix: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
unicode-ucd: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
samba: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
dbus-deviation: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
uw-imap: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
arch-install-scripts: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
openblas: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
unzip: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
sip: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
neofetch: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
wheel: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
hdparm: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
util-linux: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
iotop: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
kmscon: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
p7zip: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
subversion: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
squid: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
rhash: rebuild for global changes in R6 bring up branch

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen commented:
Packages with bumped REL
--------
```
base-devel/llvm
base-libs/libidn
base-misc/ca-certs
base-python/python-3
base-utils/bzip2
base-utils/texinfo
core-devel/gcc
extra-admin/accountsservice
extra-admin/apt
extra-admin/autofs
extra-admin/cryptsetup
extra-admin/mdadm
extra-admin/mtd-utils
extra-admin/osinfo-db-tools
extra-admin/polkit
extra-admin/quota-tools
extra-admin/systemd
extra-admin/thin-provisioning-tools
extra-admin/xfsprogs
extra-bases/desktop-base
extra-bases/python-base
extra-bases/server-base
extra-database/ldb
extra-database/tdb
extra-devel/acbs
extra-devel/cgdb
extra-devel/ftjam
extra-devel/ladspa-sdk
extra-doc/docbook-dtd
extra-editors/emacs
extra-gnome/gcr
extra-gnome/gtk-4
extra-i18n/m17n-db
extra-i18n/m17n-lib
extra-js/nodejs
extra-kernel/plymouth
extra-libs/aom
extra-libs/clucene
extra-libs/dbus-glib
extra-libs/gdk-pixbuf
extra-libs/glib-networking
extra-libs/hidapi
extra-libs/libarchive
extra-libs/libclc
extra-libs/libcork
extra-libs/libgxps
extra-libs/libiptcdata
extra-libs/libmcrypt
extra-libs/libnice
extra-libs/libotf
extra-libs/libpng
extra-libs/libsoup
extra-libs/libtasn1
extra-libs/libvdpau
extra-libs/libwmf
extra-libs/mbedtls
extra-libs/minizip
extra-libs/nss
extra-libs/qt-4
extra-libs/qt-6
extra-libs/raptor2
extra-libs/sdl2
extra-libs/serf
extra-libs/spandsp
extra-libs/talloc
extra-libs/tevent
extra-libs/yajl
extra-libs/zita-resampler
extra-misc/aosc-artwork
extra-misc/unicode-ucd
extra-multimedia/ffmpeg
extra-multimedia/gstreamer
extra-multimedia/jack
extra-multimedia/libcaca
extra-multimedia/pulseaudio
extra-network/avahi
extra-network/bird
extra-network/cifs-utils
extra-network/postfix
extra-network/ppp
extra-network/samba
extra-network/uw-imap
extra-ocaml/findlib
extra-ocaml/ocaml
extra-python/dbus-deviation
extra-python/numpy
extra-python/py3c
extra-python/pyqt5-sip
extra-python/sip
extra-python/wheel
extra-scientific/openblas
extra-utils/arch-install-scripts
extra-utils/hdparm
extra-utils/iotop
extra-utils/kmscon
extra-utils/neofetch
extra-utils/p7zip
extra-utils/rhash
extra-utils/unzip
extra-utils/util-linux
extra-vcs/subversion
extra-web/squid
```

MingcongBai committed:
ocaml: disable LTO for all architectures

MingcongBai committed:
findlib: disable LTO for all architectures

MingcongBai committed:
arch-install-scripts: update checksum for mirrorlist

MingcongBai committed:
polkit: use duktape for all architectures

Upstream defaults to Duktape over MozJS.

MingcongBai committed:
thin-provisioning-tools: fix ab_match_arch AND operator

This fixes an issue where the beyond script attempts to build Rust tools for
RISC-V, resulting in a build failure in nix.

MingcongBai committed:
zita-alsa-pcmi: (no build) use aosc-repacks; upstream tarball gone

MingcongBai committed:
zita-resampler: disable SSE2 for !amd64

MingcongBai committed:
ppp: make lib64 rename conditional

MingcongBai committed:
qt-4: (Debian patch) fix QtWebKit build on riscv64

Icenowy commented:
Said to be buildable. Will approve.

MingcongBai commented on base-devel/llvm/01-runtime/defines:
```diff
@@ -31,6 +31,7 @@ USECLANG__I486="${USECLANG__RETRO}"
 USECLANG__LOONGSON2F="${USECLANG__RETRO}"
 USECLANG__POWERPC="${USECLANG__RETRO}"
 USECLANG__PPC64="${USECLANG__RETRO}"
+USECLANG__MIPS64R6EL="${USECLANG__RETRO}"
```
Use a standalone USECLANG__MIPS64R6EL=0 switch to avoid confusion.

MingcongBai commented on base-misc/ca-certs/autobuild/build:
```diff
@@ -9,13 +9,18 @@ cp -rv "$SRCDIR"/autobuild/overrides/usr/bin/* \
 
 abinfo "Running make-ca ..."
-make-ca
+$SRCDIR/autobuild/overrides/usr/bin/make-ca --destdir "$SRCDIR"
```
Quote SRCDIR.

MingcongBai commented on core-devel/gcc/02-compiler/defines:
```diff
@@ -1,6 +1,6 @@
 PKGNAME=gcc
 PKGSEC=devel
-PKGDEP="binutils gcc-runtime gmp isl mpc mpfr texinfo zlib zstd"
+PKGDEP="binutils gcc-runtime gmp isl mpc mpfr texinfo zlib"
```
This change should have been dropped. It would have depended on zstd anyways (as it was pre-installed in BuildKit).

Move this change to `defines.stage2`.

MingcongBai commented on core-devel/gcc/spec:
```diff
@@ -1,4 +1,5 @@
 VER=12.2.0
+REL=1
```
Drop this change too.

MingcongBai commented on extra-doc/docbook-dtd/spec:
```diff
@@ -1,5 +1,6 @@
 VER=4.5
-REL=4
+REL=1
+REL=4retro1
```
Misbump?

MingcongBai commented on extra-i18n/m17n-lib/autobuild/prepare:
```diff
@@ -0,0 +1,2 @@
+abinfo "Arch Linux: use pkg-config to detect freetype2 library"
+sed -i 's/freetype-config/pkg-config freetype2/' configure
```
Absolute path here.

MingcongBai commented on extra-libs/mbedtls/autobuild/prepare:
```diff
@@ -1,8 +1,6 @@
 # FIXME: weird workaround.
 export LDFLAGS="${LDFLAGS} -I../include"
```
Missing abinfo.

MingcongBai commented on extra-ocaml/findlib/autobuild/build:
```diff
@@ -5,6 +5,8 @@ abinfo "Configuring Findlib ..."
     -mandir /usr/share/man \
     -bindir /usr/bin
 
+export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/lib/ocaml/stublibs"
```
Missing abinfo.

MingcongBai commented on extra-ocaml/ocaml/autobuild/prepare:
```diff
@@ -0,0 +1 @@
+export CFLAGS="${CFLAGS} -Wno-error=dangling-pointer="
```
Missing abinfo (is this still necessary?).

Cyanoxygen committed:
llvm: make USECLANG for mips64r6el clear

- To avoid confusion.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ca-certs: Quote variable name in build script

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
gcc: introduce stage2 receipt

- Stage 2 removed zstd from dependency.
- Reverting to previous main receipe and removing REL bump.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
docbook-dtd: fix misbump

- The correct REL should be 5.

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
m17n-lib: use absolute path in prepare script

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mbedtls: add missing abinfo in prepare script

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
findlib: add missing abinfo in build scrpit

Co-Autohred-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
ocaml: remove unncecssary compiler options in prepare

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

MingcongBai commented on base-python/python-3/autobuild/prepare:
```diff
@@ -22,3 +22,7 @@ if [[ "${CROSS:-$ARCH}" = "ppc64el" ]]; then
 fi
+if ab_match_arch mips64r6el ; then
+    abinfo "mips64r6el: Replacing multiarch as mipsisa64r6el..."
+    sed -e 's/\$CC\ --print-multiarch/echo\ mipsisa64r6el-linux-gnuabi64/g' -i configure
```
Absolute path here.

Cyanoxygen committed:
python-3: mips64r6el: switch to absolute path in prepare scrpit

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>

Cyanoxygen committed:
mbedtls: drop outdated workaround

Co-Authored-By: Mingcong Bai <jeffbai@aosc.xyz>
